tests/platform_helper/providers/test_ecs.py

```diff
@@ -176,3 +207,62 @@ def test_get_or_create_task_name_appends_random_id(mock_application):
 
     assert task_name.rsplit("-", 1)[0] == mock_task_name("app-postgres").rsplit("-", 1)[0]
     assert random_id.isalnum() and random_id.islower() and len(random_id) == 12
+
+
+@mock_aws
+def test_start_ecs_task(mocked_cluster, mock_application):
+    ecs_client = mock_application.environments["development"].session.client("ecs")
+    ssm_client = mock_application.environments["development"].session.client("ssm")
+    application_name = mock_application.name
+    env = "development"
+    mocked_ec2_client = boto3.client("ec2")
+    mocked_ec2_images = mocked_ec2_client.describe_images(Owners=["amazon"])["Images"]
+    mocked_ec2_client.run_instances(
+        ImageId=mocked_ec2_images[0]["ImageId"],
+        MinCount=1,
+        MaxCount=1,
+    )
+    mocked_ec2_instances = boto3.client("ec2").describe_instances()
+    mocked_ec2_instance_id = mocked_ec2_instances["Reservations"][0]["Instances"][0]["InstanceId"]
+
+    mocked_ec2 = boto3.resource("ec2")
+    mocked_ec2_instance = mocked_ec2.Instance(mocked_ec2_instance_id)
+    mocked_instance_id_document = json.dumps(
+        ec2_utils.generate_instance_identity_document(mocked_ec2_instance),
+    )
+
+    ecs_client.register_container_instance(
+        cluster="default",
+        instanceIdentityDocument=mocked_instance_id_document,
+    )
+
+    mocked_task_definition_arn = ecs_client.register_task_definition(
+        family="doesnt-matter",
+        containerDefinitions=[
+            {
+                "name": "test_container",
+                "image": "test_image",
+                "cpu": 256,
+                "memory": 512,
+                "essential": True,
+                "environment": [{"name": "TEST_VAR", "value": "test"}],
+            }
+        ],
+    )["taskDefinition"]["taskDefinitionArn"]
+
```

Comment:
Not personally a fan of moto, I think it adds a lot of setup complexity to the tests. Only the ecs_client is used by the method under test, and only run_task is called on that. Personally I'd just mock the client for this and do some assertions to ensure the correct parameters are passed into run_task and that returns the correct data. Not too precious about this however.

Reply:
Dragos updated the test to use unit test mocks instead of moto